guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md

```diff
@@ -0,0 +1,527 @@
+
+If you have followed our introductory guides, you have probably commented out or removed the SSL configuration, keep reading to find the template or check the repository `app.conf` as mentioned earlier.
+
+⚠️ Beware, that this is highly customizable and we might make tweaks in any earlier version of `ursa` which that guide might not consider. Check our original repository for any updates! Also, some advanced users have a preference to modify this on their own. In any case, for more on this you have to check the [Nginx docs](https://docs.nginx.com/) to learn more!
+
+2) Replace the pathname with your custom domain, by replacing the following `ursa.earth` domain with yours
+
+```sh
+# SSL code
+ssl_certificate /etc/letsencrypt/live/ursa.earth/fullchain.pem;
+ssl_certificate_key /etc/letsencrypt/live/ursa.earth/privkey.pem;
+```
+
+For our demo, we replace `ursa.earth` to `fleek-network-node.fleek.xyz`
+
+```sh
+# SSL code
+ssl_certificate /etc/letsencrypt/live/fleek-network-node.fleek.xyz/fullchain.pem;
+ssl_certificate_key /etc/letsencrypt/live/fleek-network-node.fleek.xyz/privkey.pem;
+```
+
+3) If you have followed our guides, it's likely that you have commented out or removed the lines for the SSL certificates in the Nginx `app.conf` file.
+
+Here's an example of the missing part and how it should look like (take particular care that you have to replace `<YOUR DOMAIN NAME>` and that this is only if you have deleted this part in the `app.conf` file):
+
+```sh
+server {
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+    server_name <YOUR DOMAIN NAME>;
+
+    server_tokens off;
+
+    # SSL code
+    ssl_certificate /etc/letsencrypt/live/<YOUR DOMAIN NAME>/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/<YOUR DOMAIN NAME>/privkey.pem;
+
+    include /etc/letsencrypt/options-ssl-nginx.conf;
+    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;
+
+    location /stub_status {
+      stub_status;
+    }
+
+    location / {
+      add_header content-type  application/vnd.ipld.raw;
+      add_header content-type  application/vnd.ipld.car;
+      add_header content-type  application/octet-stream;
+      add_header cache-control public,max-age=31536000,immutable;
+
+      proxy_cache nodecache;
+      proxy_cache_valid 200 31536000s;
+      add_header X-Proxy-Cache $upstream_cache_status;
+      proxy_cache_methods GET HEAD POST;
+      proxy_cache_key "$request_uri|$request_body";
+      client_max_body_size 1G;
+
+
+      proxy_pass http://ursa:4069;
+    }
+}
+```
+
+### Running the Let's Encrypt automation script
+
+First, change the directory to the `docker/full-node` path in your local repository. This is where our `init-letsencrypt.sh` is located!
+
+```sh
+cd docker/full-node
+```
+
+The directory file structure should look like:
+
+```sh
+.
+├── README.md
+├── data
+├── docker-compose.yml
+└── init-letsencrypt.sh
+```
```

Comment:
Maybe this bit of info could be better put farther above where the chmod +x is done